### DISCUSSION TITLE
刚刚通过在NAS上docker部署的alist从天翼云盘网NAS的webdav上复制数据的时候，NAS直接死机了...

### DISCUSSION THREAD

**Discussion by doraemon0711** [AlistGo/alist#4356] [General]
复制了一个文件夹，里面有3个文件，一个6G大小  
点击复制之后NAS就无法访问了（ssh等各种方式都试过），硬盘灯狂闪，当时就懵了直接重启了NAS，重启之后检查了一下没有什么大问题  
具体原因还没有排查，不知道和我没有对容器添加内存限制有没有关系

**Comment by anwen-anyi** [comment]
可能复制的时候是写入了内存，你的内存不够了？（我猜的）
你可以 按照下面的方法试试

# 如何快速定位Bug
如果发现Bug，但是log.log的日志不详细，可以尝试在 启动时候添加 `--debug` 参数启动

建议在使用`--debug` 参数启动之前将 AList目录下的日志文件/log/log.log 清空，这样方便开发者们后续快速定位问题

**Windows：alist.exe server --debug**
**Linux：./alist server --debug**
**Mac：暂时未知，可能也是可以加 --debug 参数启动**

启动后拿到相关日志，如何停止? Ctrl+c 可以使程序停止运行（或者简单粗暴直接关闭程序）

**Reply by doraemon0711** [reply]
有可能是，我限制了容器内存为512m，然后操作了一下之后容器就挂掉了，从监控来看也是内存和CPU飙升之后容器就挂掉了  
至于debug模式，我尝试了一下在docker里添加CMD的参数的方式，但貌似没生效， 可能需要改一下entrypoint.sh文件来尝试

<img width="1661" alt="屏幕截图 2023-05-10 214959" src="https://github.com/alist-org/alist/assets/13547208/e394365a-e654-4a67-af67-485d9f52ec3f">

**Reply by doraemon0711** [reply]
看来也有人遇到了同样的问题 #4149

**Reply by anwen-anyi** [reply]
`docker exec -it alist ./alist server --debug` 这样你试试呢记得先进容器把日志文件清理一下

**Reply by doraemon0711** [reply]
这是一份重现报错的日志文件，我手动对日志中的密码和地址做了一些替换（顺便将明文密码写到日志里是一件很危险的行为，不知道gin的日志框架能否对日志过滤）
[log.log](https://github.com/alist-org/alist/files/11443015/log.log)

**Reply by anwen-anyi** [reply]
> 这是一份重现报错的日志文件，我手动对日志中的密码和地址做了一些替换（顺便将明文密码写到日志里是一件很危险的行为，不知道gin的日志框架能否对日志过滤）
> 
> [log.log](https://github.com/alist-org/alist/files/11443015/log.log)
> 
> 

敏感信息自己删了就行。你是手动停止了还是报错崩溃自动停止的😁

**Reply by doraemon0711** [reply]
崩溃停止的，复制文件之后再刷新web页面就打不开了

**Reply by anwen-anyi** [reply]
😆等大佬帮你看看日志吧 不过你的内存应该是太小了 
你复制一个比你限制内存小的文件试试呢?

**Reply by doraemon0711** [reply]
小文件没问题的，我也认同你说的因为复制是写入内存从而产生问题，并且电信网使用天翼网盘本来速度就很快。

**Reply by anwen-anyi** [reply]
后面你可以等 https://github.com/alist-org/alist/commit/f97f1d532e2b76c45e66758a494873c7ee1ba0e7 这个发布了 或者你自己编译试试看还有没有 

https://github.com/alist-org/alist/commit/f97f1d532e2b76c45e66758a494873c7ee1ba0e7 修复了 https://github.com/alist-org/alist/issues/4149 的问题
